cms/djangoapps/contentstore/rest_api/v1/serializers/course_details.py

```diff
@@ -40,7 +26,7 @@ class CourseDetailsSerializer(serializers.Serializer):
     entrance_exam_enabled = serializers.CharField(allow_blank=True)
     entrance_exam_id = serializers.CharField(allow_blank=True)
     entrance_exam_minimum_score_pct = serializers.CharField(allow_blank=True)
-    instructor_info = InstructorsSerializer()
+    instructor_info = serializers.DictField(allow_empty=True)
```

Comment:
```suggestion
    instructor_info = InstructorsSerializer(allow_empty=True)
```
I think that it makes sense to keep the serializer validation so that only the expected information comes through when it is present. I agree with allowing the serializer to be empty.

Reply:
The problem I'm seeing with this is in the case when `instructor_info` isn't empty, we're not necessarily getting the `instructors` field.

I've seen the following, so I think it's possible to get `instructors` in other languages too.
```
"instructor_info": {
        "Professeurs": []
    }
```

Reply:
The MFE version of the schedule and details page does not allow for the instructor key to be saved in other languages. The frontend expects that the key is in English. The key in varying languages is a bug introduced in the legacy page because of how it handled translations.

Reply:
I see... we still have some corrupt data due to the legacy page then. If we are turning off the MFE and using legacy from time to time due to issues, would it be better to accept the corrupt data for now? When the serializer runs into an issue, it entirely shuts down course details.

Reply:
Adding `allow_empty` should fix the issue for when there is corrupt data. In the edge case that you pointed out, `Proffesseurs` is filtered out so `instructor_info` was empty. With this change, an empty dict will be sent via the API.

Reply:
Looks like that specific code doesn't work but I get the general idea. I'll look into something that will pass checks.